scripts/restructure

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/perl
+
+package PerlWeeklyChallenge;
+
+use Moo;
+use MooX::Options;
+use JSON;
+use File::Find qw(finddepth);
+use File::Basename qw(basename);
+use File::Find::Rule qw(rule);
+use File::Copy::Recursive qw(dircopy);
+
+option 'project_dir' => (is => 'ro', format => 's', doc => 'Path to the perlweeklychallenge-club folder', default => sub {"."});
+
+my $restructure_folder;
+my @weekly_challenge_folders;
+
+sub run {
+    my ($self) = @_;
+    my $project_dir = $self->{project_dir};
+    $restructure_folder = "$project_dir/restructure";
+
+    mkdir $restructure_folder or die "Failed to create folder $restructure_folder: $!";
+
+    my $members_json_file = "$project_dir/members.json";
+    @weekly_challenge_folders = rule(
+        directory =>
+        maxdepth => 1,
+        name => qr/challenge-\d{3}/,
+    )->in($project_dir);
+
+    my $members_hash_ref = read_data($members_json_file);
+    my @member_nicks = keys %$members_hash_ref;
+
+    foreach my $nick (@member_nicks)
+    {
+        populate_member_folder($nick);
+    }
+
+    print "Done";
+}
+
+sub read_data {
```

Comment:
Function like this are common and would be need many places. So I suggest put this in a separate package for example

PWC/Utils.pm